cmd/gossamer/commands/root.go

```diff
@@ -131,6 +133,10 @@ Usage:
 				}
 			}
 
+			if err := parseLogLevel(); err != nil {
```

Comment:
This must move this call before the viper configuration and validation since we're gonna use the result from this

Reply:
Ah yes, it get's overridden because `parseLogLevel()` isn't setting the viper defaults. These should help you understand: https://github.com/ChainSafe/gossamer/blob/development/cmd/gossamer/commands/utils.go#L393, https://github.com/ChainSafe/gossamer/blob/development/cmd/gossamer/commands/utils.go#L421

Reply:
that way log-level can't be set for the `init` command. because `configureViper()` and `ParseConfig()` will be called only for the `gossamer` command.

Reply:
What way exactly? I don't understand.

Reply:
there is a check [here](https://github.com/ChainSafe/gossamer/blob/1a34972b4cdfd49174bac566b3b26061fc23113f/cmd/gossamer/commands/root.go#L120) to check if it's gossamer command.
so, `configureViper()` and `ParseConfig()` will run only for gossamer command.

I think we should allow setting the log level for the `init` command also. The current code is working fine, you can test it.

Reply:
It updates in my system. attaching screenshots


before running the `gossamer init`:
![Screenshot from 2023-06-10 21-09-43](https://github.com/ChainSafe/gossamer/assets/40173579/26299923-85ea-4b0e-8934-962c1b7d17af)


after running the `gossamer init` command with   `--log core=trace`:
![Screenshot from 2023-06-10 21-11-35](https://github.com/ChainSafe/gossamer/assets/40173579/a5a826ea-fc0c-4856-870c-6d8952b8177d)


after running the `gossamer` command with `--log core=debug`:
![Screenshot from 2023-06-10 21-10-17](https://github.com/ChainSafe/gossamer/assets/40173579/4ac3818d-647f-4289-9462-bbe4e7bef285)

Reply:
Glad you got it working! I'll take a look later this week

Reply:
Just to recap, here are the changes I suggested
1. Move `parseLogLevel()` before configuring the viper. Before this [line](https://github.com/ChainSafe/gossamer/blob/development/cmd/gossamer/commands/root.go#L120)
2. Set the parsed log config to viper

As you have already noticed, the parsers are setting the parsed values in both the `config` struct and the viper store. Here is the thought behind it
- `viper` store is always the source of the config since it manages the configuration 
- the variable `config` is just an easy way for us to use the configuration without having to use constants to fetch the values via `viper.Get()`

Because of these reasons, we need to handle # 2. See [this](https://github.com/ChainSafe/gossamer/pull/3303#discussion_r1224369725) for reference

As mentioned above, viper manages the configuration for us. Here's our flow:

1. Set default config 
2. override it with the values set in flags 
3. If it is `gossamer` command, merge overridden config and the config from `config.toml` 

The priority is given to the overwritten config. 

So, if we keep it after the prerun for `gossamer` command, we are basically overriding the parsed config again; which is a bad practise. Because of this reason, we need to handle # 1

Happy to hop on a call if any of these doesn't make sense.